atc/exec/task_step.go

```diff
@@ -310,7 +309,7 @@ func (action *TaskStep) containerSpec(logger lager.Logger, repository *worker.Ar
 	} else if config.ImageResource != nil {
 		imageSpec.ImageResource = &worker.ImageResource{
 			Type:    config.ImageResource.Type,
-			Source:  creds.NewSource(action.variables, config.ImageResource.Source),
+			Source:  creds.NewSource(boshtemplate.StaticVariables{}, config.ImageResource.Source),
```

Comment:
:+1: Makes sense, since this will now be resolved as part of fetching the config.